Content for create did, replit improvements (#372)
iotaledger/identity.rs: documentation/docs/getting-started/did/intro.md

---
title: Introduction to Decentralized Identifiers
sidebar_label: Introduction
---

The Decentralized Identifiers (DID) standard from W3C is the fundamental standard that supports the concept of decentralized digital identity. A DID is a unique identifier that contains information that can be resolved to a DID Document. This document contains data such as public keys, enabling the holder to prove ownership over their personal data, but also URIs that link to public information about the identity. This implementation complies to the [DID specifications v1.0 Working](https://www.w3.org/TR/did-core//). 

In the IOTA Identity framework, we have implemented the DID standard according to the `iota` DID Method Specification, which can be viewed [here](./specs/iota_did_method_spec.md). We recommend to see the `iota` DID Method Specification as the golden standard for DID on IOTA. Other implementations of DID on IOTA are recommended to follow the `iota` DID Method Specification. However, it is not necessary to implement a novel method implementation for every project, so feel free to utilize this framework directly. 

> An example of DID conforming to the `iota` method specification:

Reviewer: ```suggestion
An example of a DID conforming to the `iota` method specification:
```

:::tip Requires basic knowledge of Asymmetric Encryption

The following and later sections require some basic knowledge of Asymmetric Encryption. Please read or view some materials on the subject before continuing. 

:::

A DID Document mostly contains two important pieces of data: public keys and services. The public keys can be used to prove ownership over the identity, by cryptographically signing something with the associated private key. The public key can be used to verify that the identity subject signed the data and therefore controls the private key. Ownership over the private keys therefore proves ownership over the identity. This also means that it is very important to keep the private keys safe and secure. In addition, the public keys allow users to send encrypted data to the identity, using their public key, that only the identity owner can decrypt. 

:::caution

Never share your private keys, seeds, passphrases with anyone. Not even IOTA Foundation members. This may lead to loss of IOTA funds or control over your own digital identity. 

:::

> Services are URIs that point to more information about the identity. This could be something as simple as a website for a organizational identity. These services are publicly available for all to read and should therefore not contain Personal Identifiable Information (PII) in the case of human identities. 

Reviewer: ```suggestion
Services are URIs that point to more information about the identity. This could be something as simple as a website for an organizational identity. These services are publicly available for all to read and should therefore not contain Personal Identifiable Information (PII) in the case of human identities. 
```

### Why use DIDs?

DIDs have some limited use cases. It allows any subject to have a unique identifier, that they can prove ownership of and at the same time provide a way to send them encrypted messages. The Identity is Self Sovereign, meaning the subject is in control of when the identity is created but also destroyed. 

> DIDs become interesting in combination with Verifiable Credentials, which will be covered in a later section. In essence, Verifiable Credentials (VCs) are signed statements by trusted third parties about a certain identity. The signer, or Issuer, is referenced by DID and so is the subject, often called the Holder. The Holder can control a copy of this statement and share this with other parties, the Verifiers, that can verify the statement and check which party gave out the statement, without having to ask the Issuer. Instead, they can verify the signature of the Issuer by checking the Issuers DID Document. This whole setup puts Holders back into control over their own data but also makes the data much more trustworthy as it has become verifiable. 

Reviewer: ```suggestion
DIDs become interesting in combination with Verifiable Credentials, which will be covered in a later section. In essence, Verifiable Credentials (VCs) are signed statements by trusted third parties about a certain identity. The signer, or Issuer, is referenced by the DID and so is the subject, often called the Holder. The Holder controls a copy of this statement and share it with other parties, the Verifiers, that can verify the statement and check which party made the statement, without having to ask the Issuer. Instead, they can verify the signature of the Issuer by checking the Issuers DID Document. This whole setup puts Holders back in control over their own data but also makes the data much more trustworthy as it has become verifiable. 
```
